Subquery Spin-Off: Prioritize query sharding (#10621)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] Distributor, querier, ingester and store-gateway: Add support for `limit` parameter for label names and values requests. #10410
* [ENHANCEMENT] Query-frontend: Allow adjustment of queries looking into the future to a maximum duration with experimental `-query-frontend.max-future-query-window` flag. #10547
* [ENHANCEMENT] Ruler: Adds support for filtering results from rule status endpoint by `file[]`, `rule_group[]` and `rule_name[]`. #10589

> * [ENHANCEMENT] Query-frontend: Add option to "spin off" subqueries as actual range queries. To enable this, set the `-query-frontend.spin-off-instant-subqueries-to-url=<url>` option on the frontend and the `instant_queries_with_subquery_spin_off` per-tenant override with regular expressions matching the queries to enable. #10460 #10603 #10621

Reviewer: [nit] Might be good to add some more detail about why someone would want to do this:

```suggestion
* [ENHANCEMENT] Query-frontend: Add option to "spin off" subqueries as actual range queries, so that they benefit from query acceleration techniques such as sharding, splitting and caching. To enable this, set the `-query-frontend.spin-off-instant-subqueries-to-url=<url>` option on the frontend and the `instant_queries_with_subquery_spin_off` per-tenant override with regular expressions matching the queries to enable. #10460 #10603 #10621
```

> * [ENHANCEMENT] Query-frontend: Add option to "spin off" subqueries as actual range queries. This is enabled by setting the `-query-frontend.spin-off-instant-subqueries-to-url=<url>` option on the frontend and the `instant_queries_with_subquery_spin_off` per-tenant override with regex expressions matching the queries to enable. #10460 #10603 #10621

Reviewer 2: ```suggestion
* [ENHANCEMENT] Query-frontend: Add option to "spin off" subqueries as actual range queries. To enable this, set the `-query-frontend.spin-off-instant-subqueries-to-url=<url>` option on the frontend and the `instant_queries_with_subquery_spin_off` per-tenant override with regular expressions matching the queries to enable. #10460 #10603 #10621
```

Author: Thank you 🙌
